torch/random.py

```diff
@@ -101,23 +101,29 @@ def initial_seed() -> int:
 
 
 @contextlib.contextmanager
-def fork_rng(devices=None, enabled=True, _caller="fork_rng", _devices_kw="devices") -> Generator:
+def fork_rng(devices=None, enabled=True, _caller="fork_rng", _devices_kw="devices", device_type="cuda") -> Generator:
```

Comment:
@albanD this API used to be very cuda-specific. Making it backend-agnostic seems reasonable, but I'm wondering why this API was never useful to CPU (and will probably break if someone were to now pass in `device_type="cpu"`). Do you happen to know?

Reply:
yeah, according to the code comment, this api is used to fork rng state for device like cuda, and it also forks the cpu rng state by default. So the device type only can be set as device type except cpu, like cuda , foo(custom device). @bdhirsh

Reply:
It is a bit weird idd.
Given that changing the current API would be painful, I think a good long term solution:
- Allow any device_type (including cpu).
- Remove the Generator return from the typing as it is not correct
- Add a `skip_cpu=False` kwarg to be able to skip cpu stashing

This can be done as a follow up as the current PR is already a working step in the right direction.

Reply:
@albanD, thanks for your reply. I will do some work according to the long-term solution you give. And now could you merge these changes?